pkg/vm/vm.go

```diff
@@ -1602,8 +1602,32 @@ func (v *VM) handleException() {
 		}
 		pop++
 		ictxv = ictxv.Next()
+		if ictxv == nil {
+			break
+		}
 		ictx = ictxv.Value().(*Context)
 	}
+	throwUnhandledException(v.uncaughtException)
+}
+
+// throwUnhandledException gets exception message from the provided stackitem and panics.
+func throwUnhandledException(item stackitem.Item) {
+	msg := "unhandled exception"
+	switch item.Type() {
+	case stackitem.ArrayT:
+		if arr := item.Value().([]stackitem.Item); len(arr) > 0 {
+			data, err := arr[0].TryBytes()
```

Comment:
And BTW, we should either use `ToString()` here or do "%q" below, because these bytes can contain anything.

Reply:
The second option is implemented.